src/app/use-cases/validate-document.ts

```diff
@@ -45,12 +45,17 @@ export type ValidateDocumentPayload = GetDocumentPayload & {
   documentUrl: string;
 };
 
+const nextStatus = (request: SignatureRequest) =>
+  request.documents.every((document) => NonEmptyString.is(document)) &&
+  request.status === "DRAFT"
+    ? "WAIT_FOR_ISSUER"
+    : request.status;
```

Comment:
you may choose to return an option and none in case of failure (or an either in case you want to store a meaningful error string)